pilot/pkg/model/push_context.go

```diff
@@ -750,12 +750,14 @@ func (ps *PushContext) UpdateMetrics() {
 }
 
 // It is called after virtual service short host name is resolved to FQDN
-func virtualServiceDestinations(v *networking.VirtualService) map[string]sets.Set[int] {
+// An ordered list of keys, and the mapping of host -> ports is returned
+func virtualServiceDestinations(v *networking.VirtualService) ([]string, map[string]sets.Set[int]) {
```

Comment:
But is this order right, we just ordered with http-> tcp->tls

Reply:
> This can cause issues in XDS stability. A fix is included

What is the influence, i couldnot understand this

Reply:
hmm tbh I am not sure it has any real world impact.. It can effect to order of cluster we create but that shouldn't matter. 

I can revert this and make the test sort the expectedServices I guess? wdyt

Reply:
yeah, sg